### PR TITLE
feat(*): API validations/cleanup, UI connection for list and delete

### PR DESCRIPTION
**Pull request by jamesfulford**
Thanks to @sarabjotanand for working with me through this.

Backend:
- Fixed API route strings (nouns, not verbs)
- Fixed the migrations (**will require destroying what we have in our "prod" db**)
- Added missing fields to model
- Separated rules by userid (filter=userid, userid=userid criteria in views), pass in via query param until auth story.
- Added more validations in serializers.
- Started backend UTs.

UI:
- Listing rules (+ UTs)
- Delete rules (+ UTs)
- Handling invalid rrule strings (at least, not exploding)
- *Actually* switched to port 8080 for UI. Oops.

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -8,4 +8,5 @@ ptvsd==4.3.2
 pymysql==0.10.1
 django-mysql==2.2.0
 djangorestframework==3.12.2
-django-cors-headers==3.3.0
\ No newline at end of file
+django-cors-headers==3.3.0
+requests==2.25.0
```
Comment (jamesfulford) [inline backend/requirements.txt:12]: For RequestsClient, used in UTs. I'm running uts with this: `docker-compose run --rm moneywise-backend python manage.py test`, though just `python manage.py test` should work if you have everything installed.

File: ui/package.json [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
-      "not op_mini all"
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
-  }
-}
+    "name": "cashflow-projection-ui",
```
Comment (jamesfulford) [inline ui/package.json:2]: We added `axios` and `axios-hooks`, everything else is auto-format.

File: ui/src/pages/plan/rules/IRule.ts [hunk 3]
```diff
@@ -10,5 +10,4 @@ export interface IApiRuleMutate {
 export interface IApiRule extends IApiRuleMutate {
     id: string;
     userid: string;
-    version: string;
```
Comment (jamesfulford) [inline ui/src/pages/plan/rules/IRule.ts:13]: I figured migrations take care of this, so I took it out. Can be added later if we really need it.

File: ui/src/pages/plan/rules/RulesContainer.tsx [hunk 4]
```diff
@@ -1,56 +1,59 @@
...
+import React, { useCallback } from 'react'
 import ListGroup from 'react-bootstrap/ListGroup';
 import { IApiRule, IApiRuleMutate } from './IRule';
 import { Rule } from './Rule';
 import {CreateForm} from './CreateRuleForm';
+import useAxios from 'axios-hooks'
+import axios from 'axios';
+
+// TODO: configure at build time
+const baseUrl = 'http://localhost:8000';
```
Comment (jamesfulford) [inline ui/src/pages/plan/rules/RulesContainer.tsx:10]: This story, but not doing tonight.

File: ui/src/pages/plan/rules/RulesContainer.tsx [hunk 5]
```diff
@@ -1,56 +1,59 @@
...
+
+// TODO: get from login
+const userid = 'test'
```
Comment (jamesfulford) [inline ui/src/pages/plan/rules/RulesContainer.tsx:13]: Later story

File: ui/src/pages/plan/rules/RulesContainer.tsx [hunk 6]
```diff
@@ -1,56 +1,59 @@
...
+        `${baseUrl}/api/rules?userid=${userid}`
+    )
+
+    const deleteHandler = useCallback((id: string) => {
+        axios.delete(`${baseUrl}/api/rules/${id}?userid=${userid}`)
+            .then(() => {
+                refetch();
+            })
+            .catch((e) => {
+                // TODO: toast an error
```
Comment (jamesfulford) [inline ui/src/pages/plan/rules/RulesContainer.tsx:26]: This story, but not tonight.

**Comment by jamesfulford** [comment]
Reviewed myself and addressed my own comments, left notes to help you guys review. (It's a practice that's helped me a lot at work)

**Review by sarabjotanand — APPROVED** [review]

**Review by rscronce — APPROVED** [review]

**Pull request merged**